src/main/java/appeng/thirdparty/codechicken/lib/model/pipeline/BakedPipeline.java

```diff
@@ -0,0 +1,434 @@
+			}
+			return clazz.cast( element.consumer );
+		}
+		throw new IllegalArgumentException( "Element with name " + name + " does not exist." );
+	}
+
+	/**
+	 * Used to enable an element on the pipeline with the specified name.
+	 *
+	 * @param name The elements name.
+	 */
+	public void enableElement( String name )
+	{
+		setElementState( name, true );
+	}
+
+	/**
+	 * Used to disable an element on the pipeline with the specified name.
+	 *
+	 * @param name The elements name.
+	 */
+	public void disableElement( String name )
+	{
+		setElementState( name, false );
+	}
+
+	/**
+	 * Used to set the state of an element on the pipeline.
+	 *
+	 * @param name The name of the element.
+	 * @param enabled The state to set it to.
+	 */
+	public void setElementState( String name, boolean enabled )
+	{
+		PipelineElement element = nameLookup.get( name );
+		if( element != null )
+		{
+			element.isEnabled = enabled;
+			return;
+		}
+		throw new IllegalArgumentException( "Element with name " + name + " does not exist." );
+	}
+
+	/**
+	 * Call when you are ready to use the pipeline.
+	 * This builds the internal state of the Elements getting things ready to transform.
+	 *
+	 * @param collector The IVertexConsumer that should collect the transformed quad.
+	 */
+	public void prepare( IVertexConsumer collector )
+	{
+		IPipelineConsumer next = null;
+		for( PipelineElement element : elements )
+		{
+			if( element.isEnabled )
+			{
+				if( first == null )
+				{
+					first = element.consumer;
+				}
+				else
+				{
+					next.setParent( element.consumer );
+				}
+				next = element.consumer;
+			}
+		}
+		next.setParent( collector );
+	}
+
+	@Override
+	public VertexFormat getVertexFormat()
+	{
+		check();
+		return first.getVertexFormat();
+	}
+
+	@Override
+	public void setQuadTint( int tint )
+	{
+		check();
+		unpacker.setQuadTint( tint );
+	}
+
+	@Override
+	public void setQuadOrientation( EnumFacing orientation )
+	{
+		check();
+		unpacker.setQuadOrientation( orientation );
+	}
+
+	@Override
+	public void setApplyDiffuseLighting( boolean diffuse )
+	{
+		check();
+		unpacker.setApplyDiffuseLighting( diffuse );
+	}
+
+	@Override
+	public void setTexture( TextureAtlasSprite texture )
+	{
+		check();
+		unpacker.setTexture( texture );
+	}
+
+	@Override
+	public void put( int element, float... data )
+	{
+		check();
+		unpacker.put( element, data );
+		if( unpacker.full )
+		{
+			onFull();
+		}
+	}
+
+	@Override
+	public void put( Quad quad )
+	{
+		check();
+		unpacker.put( quad );
+	}
+
+	private void check()
+	{
+		if( first == null )
+		{
+			throw new IllegalStateException( "Pipeline used before prepare was called." );
+		}
+	}
+
+	private void onFull()
+	{
+		first.setInputQuad( unpacker );
+		first.put( unpacker );
+	}
+
+	/**
+	 * Internal class, used to hold a PipelineElement's state.
+	 */
+	public static class PipelineElement<T extends IPipelineConsumer>
+	{
+
+		public String name;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make name a static final constant or non-public and provide accessors if needed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)